### PR TITLE
Prepare for Transfer

### PR DESCRIPTION
**Pull request by MBoegers**
see rewrite.yml which utilizes OpenRewrite to migrate Packages, Classnames and properties. 

Fix for #35

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,15 +1,15 @@
-# Hedera Enterprise
-This project provides Java modules to integrate [Hedera network](https://hedera.com) smoothless in a Spring Boot or Microprofile (like Quarkus) application.
-This module is based on the [Hedera Java SDK](https://github.com/hashgraph/hedera-sdk-java).
+# Hiero Enterprise
+This project provides Java modules to integrate [Hiero network](https://hiero.com) smoothless in a Spring Boot or Microprofile (like Quarkus) application.
```
Comment (hendrikebbers) [inline README.md:2]:
```suggestion
This project provides Java modules to integrate [Hiero network](https://hiero.org) smoothless in a Spring Boot or Microprofile (like Quarkus) application.
```

File: README.md [hunk 2]
```diff
@@ -1,15 +1,15 @@
...
+This module is based on the [Hiero Java SDK](https://github.com/hashgraph/hiero-sdk-java).
```
Comment (hendrikebbers) [inline README.md:3]:
```suggestion
This module is based on the [Hiero Java SDK](https://github.com/hashgraph/hedera-sdk-java).
```

File: README.md [hunk 3]
```diff
@@ -19,40 +19,40 @@ To use this module, you need to add the following dependency to your project:
...
-spring.hedera.network=testnet
+spring.hiero.accountId=0.0.53854625
+spring.hiero.privateKey=2130020100312346052b8104400304220420c236508c429395a8180b1230f436d389adc5afaa9145456783b57b2045c6cc37
+spring.hiero.network=testnet
 ```
 
 The account information (accountId, privateKey, publicKey) can all be found at the
-[Hedera portal](https://portal.hedera.com/) for a testnet or previewnet account.
-Today only the "DER Encoded Private Key" of the "ECDSA" key type is supported for the `spring.hedera.privateKey` property.
+[Hiero portal](https://portal.hiero.com/) for a testnet or previewnet account.
```
Comment (hendrikebbers) [inline README.md:28]:
```suggestion
[Hedera portal](https://portal.hedera.com/) for a testnet or previewnet account.
```

File: README.md [hunk 4]
```diff
@@ -19,40 +19,40 @@ To use this module, you need to add the following dependency to your project:
...
+Today only the "DER Encoded Private Key" of the "ECDSA" key type is supported for the `spring.hiero.privateKey` property.
 
-The 2 properties `spring.hedera.accountId` and `spring.hedera.privateKey` define the so called "operation account".
-The operational account is used as the account that sends all transactions against the Hedera network.
+The 2 properties `spring.hiero.accountId` and `spring.hiero.privateKey` define the so called "operation account".
+The operational account is used as the account that sends all transactions against the Hiero network.
```
Comment (hendrikebbers) [inline README.md:32]:
```suggestion
The operational account is used as the account that sends all transactions against a Hiero network.
```

File: README.md [hunk 5]
```diff
@@ -19,40 +19,40 @@ To use this module, you need to add the following dependency to your project:
...
+@EnableHiero
 public class Application {
     public static void main(String[] args) {
         SpringApplication.run(Application.class, args);
     }
 }
 ```
  
-Once that is done you can for example autowire the `FileClient` class and call the methods to interact with the Hedera network.
+Once that is done you can for example autowire the `FileClient` class and call the methods to interact with the Hiero network.
```
Comment (hendrikebbers) [inline README.md:50]:
```suggestion
Once that is done you can for example autowire the `FileClient` class and call the methods to interact with a Hiero based network.
```

File: README.md [hunk 6]
```diff
@@ -68,16 +68,16 @@ public class HederaAccountService {
...
-### Hedera Spring Sample
+### Hiero Spring Sample
 
-A sample application that uses the Hedera Spring module can be found in the `hedera-spring-sample` module.
+A sample application that uses the Hiero Spring module can be found in the `hiero-spring-sample` module.
 The sample application is a simple Spring Boot application that reads has a REST endpoint at `localhost:8080/` and 
 shows the hbar balance of the account `0.0.100`.
-To use the application, you need to have created a Hedera testnet account at the [Hedera portal](https://portal.hedera.com/).
-The account information can be added to the `application.properties` file in the `hedera-spring-sample` module:
+To use the application, you need to have created a Hiero testnet account at the [Hiero portal](https://portal.hiero.com/).
```
Comment (hendrikebbers) [inline README.md:76]:
```suggestion
To use the application, you need to have created a Hedera testnet account at the [Hedera portal](https://portal.hedera.com/).
```

File: README.md [hunk 7]
```diff
@@ -92,20 +92,20 @@ The support for Microprofile is still in development and can not be used yet.
 
 ## Managed services
 
-The module provides a set of managed services that can be used to interact with the Hedera network.
+The module provides a set of managed services that can be used to interact with the Hiero network.
```
Comment (hendrikebbers) [inline README.md:95]:
```suggestion
The module provides a set of managed services that can be used to interact with a Hiero based network.
```

File: README.md [hunk 8]
```diff
@@ -115,7 +115,7 @@ The project is based on [Maven](https://maven.apache.org/). To build the project
 ./mvnw verify
 ```
 
-The tests in the project are working against the Hedera testnet.
+The tests in the project are working against the Hiero testnet.
```
Comment (hendrikebbers) [inline README.md:118]:
```suggestion
The tests in the project are working against the Hedera testnet.
```

File: hedera-base/src/main/java/module-info.java [hunk 9]
```diff
@@ -1,10 +0,0 @@
-module com.openelements.hedera.base {
```
Comment (hendrikebbers) [inline hedera-base/src/main/java/module-info.java:1]: The file should not be removed

File: hedera-spring/src/test/resources/application.properties [hunk 10]
```diff
@@ -1,6 +0,0 @@
-spring.config.import=optional:file:.env[.properties]
```
Comment (hendrikebbers) [inline hedera-spring/src/test/resources/application.properties:1]: The file should not be deleted

File: hiero-base/src/main/java/module-info.java [hunk 11]
```diff
@@ -0,0 +1,10 @@
+module com.openelements.hiero.base {
```
Comment (hendrikebbers) [inline hiero-base/src/main/java/module-info.java:1]: lol, ok. er hat icht gecheckt, dass die Datei verschoben wurde....

File: hiero-base/src/test/java/com/openelements/hiero/base/test/HieroTestContext.java [hunk 12]
```diff
@@ -13,15 +13,15 @@
...
-    private static final Logger log = LoggerFactory.getLogger(HederaTestContext.class);
+    private static final Logger log = LoggerFactory.getLogger(HieroTestContext.class);
 
     private final Account operationalAccount;
 
     private final Client client;
 
-    public HederaTestContext() {
+    public HieroTestContext() {
         final String hederaAccountIdByEnv = System.getenv("HEDERA_ACCOUNT_ID");
```
Comment (hendrikebbers) [inline hiero-base/src/test/java/com/openelements/hiero/base/test/HieroTestContext.java:25]:
```suggestion
        final String hederaAccountIdByEnv = System.getenv("HIERO_ACCOUNT_ID");
```

File: hiero-base/src/test/java/com/openelements/hiero/base/test/HieroTestContext.java [hunk 13]
```diff
@@ -13,15 +13,15 @@
...
         final String hederaPrivateKeyByEnv = System.getenv("HEDERA_PRIVATE_KEY");
```
Comment (hendrikebbers) [inline hiero-base/src/test/java/com/openelements/hiero/base/test/HieroTestContext.java:26]:
```suggestion
        final String hederaPrivateKeyByEnv = System.getenv("HIERO_PRIVATE_KEY");
```

File: hiero-base/src/test/java/com/openelements/hiero/base/test/HieroTestContext.java [hunk 14]
```diff
@@ -13,15 +13,15 @@
...
         final String hederaNetwork = System.getenv("HEDERA_NETWORK");
```
Comment (hendrikebbers) [inline hiero-base/src/test/java/com/openelements/hiero/base/test/HieroTestContext.java:27]:
```suggestion
        final String hederaNetwork = System.getenv("HIERO_NETWORK");
```

**Comment by MBoegers** [comment]
I'd like to see the content of `rewrite.yml` be distributed as RecipeArtifact for OpenRewrite to pave the migration path from hedera to hiero for downstream users. The resulting migration could be contributed to [OpenRewrite/rewrite-third-parts](https://github.com/openrewrite/rewrite-third-party) to be easily accessible via the OpenRewrite Recipe catalog. Is there any initiative — yet?

File: hedera-base/src/main/java/module-info.java (hunk 9 above)
Comment (MBoegers) [reply hedera-base/src/main/java/module-info.java:1]: it's still there ;)

File: hiero-base/src/main/java/module-info.java (hunk 11 above)
Comment (MBoegers) [reply hiero-base/src/main/java/module-info.java:1]: :D

File: hedera-spring/src/test/resources/application.properties (hunk 10 above)
Comment (MBoegers) [reply hedera-spring/src/test/resources/application.properties:1]: I think its still there

**Comment by hendrikebbers** [comment]
@MBoegers, I will add the idea of OpenRewrite receipts as a topic for our next TSC meeting. It is a good idea to have support for our SDKs.

**Comment by hendrikebbers** [comment]
Added at https://github.com/hiero-ledger/tsc/blob/main/minutes/2024-12-17.md. I will close this PR for now.

**Pull request closed without merging**